File: viking-processor/src/main/java/net/droidlabs/viking/processor/annotation/AnnotationUtil.java
package net.droidlabs.viking.processor.annotation;

import java.util.ArrayList;
import java.util.List;
import java.util.Map;
import javax.lang.model.type.TypeMirror;

import net.droidlabs.viking.processor.AnnotatedClass;
import net.droidlabs.viking.annotations.AutoModule;

import static net.droidlabs.viking.processor.Util.getAnnotation;

public final class AnnotationUtil {
  private AnnotationUtil() {
  }

  public static List<TypeMirror> getValuesForAttribute(String attribute,
      AnnotatedClass annotatedClass) {
    Map<String, Object> parsedAnnotation = getAnnotation(AutoModule.class,
        annotatedClass.getTypeElement());

    Object[] types = (Object[]) parsedAnnotation.get(attribute);
    List<TypeMirror> typeMirrors = new ArrayList<>();
    for (Object type : types) {
      TypeMirror typeMirror = (TypeMirror) type;
      typeMirrors.add(typeMirror);
    }

    return typeMirrors;
  }
}

File: viking-processor/src/main/java/net/droidlabs/viking/processor/annotation/AnnotationUtil.kt
package net.droidlabs.viking.processor.annotation

import net.droidlabs.viking.annotations.AutoModule
import net.droidlabs.viking.processor.AnnotatedClass
import net.droidlabs.viking.processor.Util.getAnnotation
import java.util.*
import javax.lang.model.type.TypeMirror

object AnnotationUtil {

    fun getValuesForAttribute(attribute: String,
                              annotatedClass: AnnotatedClass<*>): List<TypeMirror> {
        val parsedAnnotation = getAnnotation(AutoModule::class.java,
                annotatedClass.typeElement2)

        val types = parsedAnnotation!![attribute] as Array<Any>
        val typeMirrors = ArrayList<TypeMirror>()
        for (type in types) {
            val typeMirror = type as TypeMirror
            typeMirrors.add(typeMirror)
        }

        return typeMirrors
    }
}
